### PR TITLE
chore: Differentiate 4xx Http errors from an actual server error

### PR DESCRIPTION
**Pull request by lutangar**
I tried not to impact the current behavior but just to change the 4xx error behavior.

> see #360 for history

### CODE REVIEW

**Comment by lutangar** [comment]
ping @JalilArfaoui

**Comment by lutangar** [comment]
ping @JalilArfaoui

File: src/api/call.ts [hunk 1]
```diff
@@ -47,12 +58,11 @@ export const post = (path: string, data: {} | []) =>
...
-  }).then(response => {
-    if (response.status >= 400) {
-      throw new Error('Bad response from server');
-    }
-    if (response.status === 204) {
-      return true;
-    }
-    return response.json();
-  });
+  })
```
Comment (JalilArfaoui) [inline src/api/call.ts:61]: Why only on `post` calls ?

File: src/api/call.ts [hunk 2]
```diff
@@ -39,6 +39,17 @@ export const get = (path: string, data: object = {}) => {
   );
 };
 
+export const checkStatus = (response: Response) => {
+  const { status } = response;
+  if (status >= 200 && status < 300) {
+    return response;
+  } else if (status >= 500) {
+    throw new Error('Bad response from server');
```
Comment (JalilArfaoui) [inline src/api/call.ts:47]: Maybe we could use `APIStatusCodeError` ?

Comment (lutangar) [reply src/api/call.ts:47]: Sure, I updated `APIStatusCodeError` to use statusText as well.

File: src/api/call.ts (hunk 1 above)
Comment (lutangar) [reply src/api/call.ts:61]: Behavior is now the same for both.

**Review by JalilArfaoui — APPROVED** [review]

**Comment by JalilArfaoui** [comment]
@lutangar so sorry for my review delay !

**Comment by JalilArfaoui** [comment]
This is conflicting new

**Pull request merged**